Sources/Widgets/Widgets3D/ResliceCursorWidget/test/testMultipleRotationPlanes.js

```diff
@@ -0,0 +1,373 @@
+            resetFocalPoint: false,
+            keepFocalPointPosition,
+            computeFocalPointOffset: false,
+            resetViewUp: false,
+          });
+          obj.renderWindow.render();
+        });
+      }
+
+      /**
+       * Update the reslice view and compare it to ground truthes values
+       * @param {Number} viewId Defines the id of compared view. Id is used to get the correct viewAttributes object
+       * @param {Object} comparisonValues Contains ground truthes values that will be compared (focalPoint, viewUp, origin, point1, point2)
+       */
+      function updateAndCompareView(viewId, comparisonValues) {
+        const viewObj = viewAttributes[viewId];
+        const out = updateReslice({
+          viewType: sliceTypes[viewId],
+          reslice: viewObj.reslice,
+          actor: viewObj.resliceActor,
+          renderer: viewObj.renderer,
+          resetFocalPoint: false,
+          keepFocalPointPosition: true,
+          computeFocalpointOffset: false,
+          resetViewUp: false,
+        });
+        const tolerance = 1e-4;
+        const camera = viewObj.renderer.getActiveCamera();
+        t.ok(
+          vtkMath.areEquals(
+            camera.getFocalPoint(),
+            comparisonValues.focalPoint,
+            tolerance
+          )
+        );
+        t.ok(
+          vtkMath.areEquals(
+            camera.getViewUp(),
+            comparisonValues.viewUp,
+            tolerance
+          )
+        );
+        t.ok(vtkMath.areEquals(out.origin, comparisonValues.origin, tolerance));
+        t.ok(vtkMath.areEquals(out.point1, comparisonValues.point1, tolerance));
+        t.ok(vtkMath.areEquals(out.point2, comparisonValues.point2, tolerance));
+      }
+
+      // ----------------------------------------------------------------------
+      // Check initialization
+      // Check X view
+      updateAndCompareView(1, {
+        focalPoint: [179.296875, 179.296875, 165],
+        viewUp: [0, 0, 1],
+        origin: [179.296875, 0, 0],
+        point1: [179.296875, 358.59375, 0],
+        point2: [179.296875, 0, 330],
+      });
+      // Check Y view
+      updateAndCompareView(0, {
+        focalPoint: [179.296875, 179.296875, 165],
+        viewUp: [0, 0, 1],
+        origin: [0, 179.296875, 0],
+        point1: [358.59375, 179.296875, 0],
+        point2: [0, 179.296875, 330],
+      });
+      // Check Z view
+      updateAndCompareView(2, {
+        focalPoint: [179.296875, 179.296875, 165],
+        viewUp: [0, -1, 1.5851759508024892e-16],
+        origin: [0, 358.59375, 165],
+        point1: [358.59375, 358.59375, 165],
+        point2: [0, 9.466330862652142e-30, 165.00000000000006],
+      });
+
+      // ----------------------------------------------------------------------
+      // Test rotate Z around Y
+      widget.rotatePlane(
+        'X',
+        vtkMath.radiansFromDegrees(44.99999999999998),
+        'Y'
+      );
+      updateViews(true);
```

Comment:
why is it needed ? `updateAndCompareView` will call `updateReslice` on the views also.

Reply:
Indeed, here we don't need it as we test three views. If we had tested less than the three views, then we'll have to update  the whole views. Else, the viewup computation will fail.